src/Elliptic/Systems/Xcts/HydroQuantities.cpp

```diff
@@ -0,0 +1,23 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Elliptic/Systems/Xcts/HydroQuantities.hpp"
+
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "PointwiseFunctions/GeneralRelativity/IndexManipulation.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace Xcts::Tags {
+
+void LowerSpatialFourVelocityCompute::function(
```

Comment:
The SpECTRE GRMHD code actually uses `Wv^i` since we always already have `W` and `v^i`. The evolution code has no support for `u_i`. The reason is that everything always works in `W` and `v^i` so it's easier to just use `Wv^i` than `u_i=Wgamma_{ij}v^j`. I don't know why SpEC uses `u_i`, since `Wv^i` also guarantees a causal spatial velocity if interpolated. In summary: if you're observing this for the SpECTRE evolution code, you should do `Wv^i` since that's what the code uses. There's even a tag for it: https://github.com/sxs-collaboration/spectre/blob/9c14163b07470050cee18ea1a4e64bceb56886ec/src/PointwiseFunctions/Hydro/Tags.hpp#L212

Reply:
Where is `LorentzFactorTimesSpatialVelocity` added to the DataBox? I can see how the ValenciaDivClean system specifies primitive vars which are retrieved from the analytic solution / data, but it's not transparent to me how `W v^i` gets into the DataBox.

Reply:
It's only used in the reconstruction, it's not added to the DataBox. What I'm saying is that tag exists and since we already use `Wv^i` it might be less confusing to just stick to `Wv^i` and avoid adding `u_i`, if possible :)

Reply:
@ffoucart any preference for interpolating `W v^i` vs `W gamma_ij v^j`?

Reply:
(@nilsdeppe this quantity is only written out of the ID solver, then interpolated, then split into `W` and `v^i` and stored in the DataBox. So I think it doesn't necessarily have to be the same as an evolution / reconstruction quantity. I have no preference here, I think we should just pick whichever is best for interpolation / storage. Maybe we'll have to experiment with that a bit as well)

Reply:
Okay, probably the only thing that would be good is to make both SpEC and SpECTRE dump the same so it's easier to deal with. I have no preference overall.

Reply:
Yes, just do `u_i` for now. No need to try to figure if one is slightly better than the other :) Probably doesn't matter since they only differ by the spatial metric, which should be relatively smooth

Reply:
This PR is ready to merge then I think